subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -728,12 +729,23 @@ static uint8_t adv_sync_hdr_set_clear(struct lll_adv_sync *lll_sync,
 		ter_dptr++;
 	}
 
-	/* TODO: ACAD */
-
-	/* AdvData */
-	/* Calc previous tertiary PDU len */
-	ter_len_prev = ull_adv_aux_hdr_len_calc(ter_com_hdr_prev,
-						&ter_dptr_prev);
+	/* Calc previous ACAD len and update PDU len */
+	ter_len_prev = ter_dptr_prev - (uint8_t *)ter_com_hdr_prev;
+	if (ter_len_prev < (ter_com_hdr_prev->ext_hdr_len +
+			    offsetof(struct pdu_adv_com_ext_adv,
+				     ext_hdr_adv_data))) {
+		acad_len_prev = ter_com_hdr_prev->ext_hdr_len +
+				offsetof(struct pdu_adv_com_ext_adv,
+					 ext_hdr_adv_data) - ter_len_prev;
+		ter_len_prev += acad_len_prev;
+		ter_dptr_prev += acad_len_prev;
+		ter_dptr += acad_len_prev;
+	} else {
+		acad_len_prev = 0;
+		ter_len_prev = offsetof(struct pdu_adv_com_ext_adv,
+					ext_hdr_adv_data);
+		ter_dptr_prev = (uint8_t *)ter_com_hdr_prev + ter_len_prev;
```

Comment:
Why are `ter_len_prev` and `ter_dptr_prev` set here instead of keeping their values?

Reply:
Added comments in code.